File: src/slam_robot_py/slam_robot_py/cmd_vel_listener.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from geometry_msgs.msg import Twist

class CmdVelListener(Node):
    def __init__(self):
        super().__init__('cmd_vel_listener_py')
        self.sub = self.create_subscription(Twist, '/cmd_vel', self.cb, 10)

    def cb(self, msg: Twist):
        self.get_logger().info(f"CmdVel (py): linear.x={msg.linear.x:.3f} angular.z={msg.angular.z:.3f}")

def main(args=None):
    rclpy.init(args=args)
    node = CmdVelListener()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: src/slam_robot_cpp/src/cmd_vel_listener.cpp
#include <memory>
#include "rclcpp/rclcpp.hpp"
#include "geometry_msgs/msg/twist.hpp"

class CmdVelListener : public rclcpp::Node {
public:
  CmdVelListener(): Node("cmd_vel_listener") {
    sub_ = this->create_subscription<geometry_msgs::msg::Twist>(
      "/cmd_vel", 10,
      [this](geometry_msgs::msg::Twist::UniquePtr msg) {
        RCLCPP_INFO(this->get_logger(),
                    "CmdVel received: linear x=%.3f angular z=%.3f",
                    msg->linear.x, msg->angular.z);
      });
  }
private:
  rclcpp::Subscription<geometry_msgs::msg::Twist>::SharedPtr sub_;
};

int main(int argc, char **argv) {
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<CmdVelListener>());
  rclcpp::shutdown();
  return 0;
}
